### ISSUE TITLE
Detach mode prints full output

### ISSUE THREAD

**Issue by qhenkart** [docker/compose#10909]
### Description

on all previous versions of docker, I had a system like this

docker-compose up -d redis
then I would run
docker-compose up server

previously the output of redis would not show up in the terminal window where the server is running. Since upgrading, it now does

### Steps To Reproduce

1. docker-compose up -d <some-container-with-noisy-output>
2. docker-compose up <some-container-that-depends-on-the-detached-container>
3. see output of 1 show up in the output terminal of 2

### Compose Version

```Text
Docker Compose version v2.20.2-desktop.1
```

### Docker Environment

```Text
Client:
 Version:    24.0.5
 Context:    desktop-linux
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.11.2-desktop.1
    Path:     /Users/qhenkart/.docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.20.2-desktop.1
    Path:     /Users/qhenkart/.docker/cli-plugins/docker-compose
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.0
    Path:     /Users/qhenkart/.docker/cli-plugins/docker-dev
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.20
    Path:     /Users/qhenkart/.docker/cli-plugins/docker-extension
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v0.1.0-beta.6
    Path:     /Users/qhenkart/.docker/cli-plugins/docker-init
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     /Users/qhenkart/.docker/cli-plugins/docker-sbom
  scan: Docker Scan (Docker Inc.)
    Version:  v0.26.0
    Path:     /Users/qhenkart/.docker/cli-plugins/docker-scan
  scout: Command line tool for Docker Scout (Docker Inc.)
    Version:  0.20.0
    Path:     /Users/qhenkart/.docker/cli-plugins/docker-scout

Server:
 Containers: 44
  Running: 41
  Paused: 0
  Stopped: 3
 Images: 8
 Server Version: 24.0.5
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 3dce8eb055cbb6872793272b4f20ed16117344f8
 runc version: v1.1.7-0-g860f061
 init version: de40ad0
 Security Options:
  seccomp
   Profile: unconfined
  cgroupns
 Kernel Version: 5.15.49-linuxkit-pr
 Operating System: Docker Desktop
 OSType: linux
 Architecture: aarch64
 CPUs: 5
 Total Memory: 13.4GiB
 Name: docker-desktop
 ID: 053f3819-0c0d-4f1d-b576-3d73dab53590
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: daemon is not using the default seccomp profile
```
```


### Anything else?

_No response_

**Comment by qhenkart** [comment]
@milas I think your PR fixed this. But just to give more context. The missing piece of my report was that it only happens if "server" used depends_on: redis in the docker-compose file

**Comment by milas** [comment]
Thanks for the follow-up 🙂 

It's included in [v2.21.0](https://github.com/docker/compose/releases/tag/v2.21.0), let me know if you see any other issues here!
